src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -137,4 +142,56 @@
         }
         return new ProjectName(trimmedName);
     }
+
+    /**
+     * Parses a String into a description
+     *
+     * @param description
+     * @return description.
+     * @throws ParseException exception related.
+     */
+    public static String parseDescription(String description) throws ParseException {
+        requireNonNull(description);
+        String trimmedDescription = description.trim();
+        if (!trimmedDescription.matches("[\\p{Alnum}][\\p{Alnum} ]*")) {
+            throw new ParseException(Event.MESSAGE_DESCRIPTION_CONSTRAINTS);
+        }
+        return trimmedDescription;
+    }
+
+    /**
+     * Parses a String into an interval enum.
+     *
+     * @param interval
+     * @return Interval enum.
+     * @throws ParseException exception related.
+     */
+    public static Interval parseInterval(String interval) throws ParseException {
+        requireNonNull(interval);
+        String trimmedInterval = interval.trim();
+        try {
+            Interval.valueOf(trimmedInterval);
+        } catch (IllegalArgumentException iae) {
+            throw new ParseException(Event.MESSAGE_INTERVAL_CONSTRAINTS);
+        }
+        return Interval.valueOf(trimmedInterval);
+    }
+
+    /**
+     * Parses a String into a localDate.
+     *
+     * @param date
+     * @return localDate.
+     * @throws ParseException exception related.
+     */
+    public static LocalDate parseDate(String date) throws ParseException {
+        requireNonNull(date);
+        String trimmedDate = date.trim();
+        try {
+            LocalDate.parse(trimmedDate, DateTimeFormatter.ofPattern("dd-MM-uuuu"));
+        } catch (DateTimeParseException e) {
+            throw new ParseException(Event.MESSAGE_DATE_CONSTRAINTS);
+        }
+        return LocalDate.parse(trimmedDate, DateTimeFormatter.ofPattern("dd-MM-uuuu"));
```

Comment:
same here, return inside try catch block.

Let's use yyyy instead of uuuu for better compatibility (with for e.g. simple date parser.)

Reply:
resolved